util/check_tool_requirements.py

```diff
@@ -23,6 +25,261 @@ def get_tool_requirements_path():
+        except ValueError as err:
+            return (False,
+                    'Failed to convert requirement to semantic version: {}'
+                    .format(err))
+        try:
+            min_sv = StrictVersion(min_semver)
+        except ValueError as err:
+            return (False,
+                    'Bad semver inferred from required version ({}): {}'
+                    .format(min_semver, err))
+
+        try:
+            actual_version = self.get_version()
+        except RuntimeError as err:
+            return (False, str(err))
+
+        try:
+            actual_semver = self.to_semver(actual_version)
+        except ValueError as err:
+            return (False,
+                    'Failed to convert installed to semantic version: {}'
+                    .format(err))
+        try:
+            actual_sv = StrictVersion(actual_semver)
+        except ValueError as err:
+            return (False,
+                    'Bad semver inferred from installed version ({}): {}'
+                    .format(actual_semver, err))
+
+        if actual_sv < min_sv:
+            return (False,
+                    'Installed version is too old: '
+                    'found version {}, but need at least {}'
+                    .format(actual_version, self.min_version))
+
+        return (True,
+                'Sufficiently recent version (found {}; needed {})'
+                .format(actual_version, self.min_version))
+
+
+class VerilatorToolReq(ToolReq):
+    def get_version(self):
+        try:
+            # Note: "verilator" needs to be called through a shell and with all
+            # arguments in a string, as it doesn't have a shebang, but instead
+            # relies on perl magic to parse command line arguments.
+            version_str = subprocess.run('verilator --version', shell=True,
+                                         check=True, stdout=subprocess.PIPE,
+                                         stderr=subprocess.STDOUT,
+                                         universal_newlines=True)
+        except subprocess.CalledProcessError as err:
+            raise RuntimeError('Unable to call Verilator to check version: {}'
+                               .format(err)) from None
+
+        return version_str.stdout.split(' ')[1].strip()
+
+
+class VeribleToolReq(ToolReq):
+    tool_cmd = ['verible-verilog-lint', '--version']
+
+    def to_semver(self, version):
+        # Drop the hash suffix and convert into version string that
+        # is compatible with StrictVersion in check_version below.
+        # Example: v0.0-808-g1e17daa -> 0.0.808
+        m = re.fullmatch(r'v([0-9]+)\.([0-9]+)-([0-9]+)-g[0-9a-f]+$', version)
+        if m is None:
+            raise ValueError("{} has invalid version string format."
+                             .format(version))
+
+        return '.'.join(m.group(1, 2, 3))
+
+
+class VcsToolReq(ToolReq):
+    tool_cmd = ['vcs', '-full64', '-ID']
```

Comment:
This does not work at our end unfortunately. We will get back to you on this.

Reply:
```suggestion
    tool_cmd = ['VCS_ARCH_OVERRIDE=linux', 'vcs', '-full64', '-ID']
```

The problem is the use of non-standard linux kernel on Google machines that is causing this to fail. The fix suggested above works.

Reply:
Hmm, that's weird: I'm happily running VCS on Debian unstable here (which I'm pretty certain Synopsys doesn't support :-)). But, yep, I can add that.

Reply:
These environment variables should really be set by the environment manager, e.g. Environment Modules, where also things like VCS_HOME are set. Another company will require different settings. It's a shame that even the `-full64` is required.

@sriyerg can you check if VCS_ARCH_OVERRIDE is already set in your environment when you load VCS, and if not, ask IT to add it to the environment?

Reply:
> These environment variables should really be set by the environment manager, e.g. Environment Modules, where also things like VCS_HOME are set.

I agree, but this hack is already in our vcs.hjson (I checked). I propose we get this in now, since it doesn't make things any worse. Maybe we can strengthen our requirements ("a working VCS installation") in the future.